### PR TITLE
support gather_facts: false; support setup-snapshot.yml

### PR DESCRIPTION
**Pull request by richm**
Some users use `gather_facts: false` in their playbooks.  This changes
the role to work in that case, by gathering only the facts it requires
to run.
CI testing can be sped up by creating a snapshot image pre-installed
with packages.  tests/setup-snapshot.yml can be used by a CI system
to do this.

### CODE REVIEW

File: tests/setup-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+- hosts: all
```
Comment (martinpitt) [inline tests/setup-snapshot.yml:1]: This should be a separate commit, as it's unrelated to the facts gathering.

File: tasks/main.yml [hunk 2]
```diff
@@ -1,11 +1,5 @@
-- name: Set version specific variables
-  include_vars: "{{ item }}"
-  with_first_found:
-    - "{{ ansible_distribution }}-{{ ansible_distribution_version }}.yml"
-    - "{{ ansible_distribution }}-{{ ansible_distribution_major_version }}.yml"
-    - "{{ ansible_distribution }}.yml"
-    - "{{ ansible_os_family }}.yml"
-    - "default.yml"
+- name: Ensure ansible_facts and variables used by role
+  include_tasks: tasks/set_vars.yml
```
Comment (martinpitt) [inline tasks/main.yml:2]: Hmm, so `include_role:` requires explicit `public: true` to expose its variables. But [include_tasks](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_tasks_module.html) does not have such a flag at all. Looks oddly asymmetric, but apparently works.

Just thinking aloud, nothing to change here.

**Review by martinpitt — CHANGES_REQUESTED** [review]
Tests [fail](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/linux-system-roles-cockpit-pull-linux-system-roles_cockpit-58-f598902-fedora-35-20220413-012011/artifacts/ansible.log) on 

    The task includes an option with an undefined variable. The error was: '__certificate_default_directory' is undefined

This variable is defined in the [lsr-certificate role](https://github.com/linux-system-roles/certificate/blob/master/vars/main.yml#L22). So somehow this PR now hides the variable from the test playbook?

Thank you!

File: tasks/main.yml (hunk 2 above)
Comment (richm) [reply tasks/main.yml:2]: > Hmm, so `include_role:` requires explicit `public: true` to expose its variables. But [include_tasks](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/include_tasks_module.html) does not have such a flag at all. Looks oddly asymmetric, but apparently works.

A task file cannot create variables that are private to the file.  A role can create variables (in vars/) that are private to the role.

> 
> Just thinking aloud, nothing to change here.

**Comment by richm** [comment]
> Tests [fail](https://dl.fedoraproject.org/pub/alt/linuxsystemroles/logs/linux-system-roles-cockpit-pull-linux-system-roles_cockpit-58-f598902-fedora-35-20220413-012011/artifacts/ansible.log) on
> 
> ```
> The task includes an option with an undefined variable. The error was: '__certificate_default_directory' is undefined
> ```
> 
> This variable is defined in the [lsr-certificate role](https://github.com/linux-system-roles/certificate/blob/master/vars/main.yml#L22). So somehow this PR now hides the variable from the test playbook?

Not sure.  But the way that test was written with multiple plays - call the certificate role in one play, and use the variable from it in another play - I'm not sure why this ever worked.  It is possible that this wasn't supposed to be supported, and Ansible was changed to "fix the glitch".  This has happened before in Ansible when undocumented behavior has changed in a minor update.  There is a commit in this PR which fixes that - use `public: true` to explicitly export the certificate private variables into the play, and run all of the test tasks in a single play so that subsequent tasks can use the variables exported by the certificate role include_role task.

> 
> Thank you!

**Comment by richm** [comment]
[citest pending]

**Review by martinpitt — COMMENTED** [review]
Thanks @richm ! LGTM, sorry for the delay (PTO)
